packages/workbox-broadcast-cache-update/BroadcastCacheUpdate.mjs

```diff
@@ -75,24 +154,33 @@ class BroadcastCacheUpdate {
   }
 
   /**
-   * Compare two [Responses](https://developer.mozilla.org/en-US/docs/Web/API/Response)
-   * and send a message via the
-   * {@link https://developers.google.com/web/updates/2016/09/broadcastchannel|Broadcast Channel API}
-   * if they differ.
-   *
-   * Neither of the Responses can be {@link http://stackoverflow.com/questions/39109789|opaque}.
+   * Waits for a message from the window indicating that it's capable of
+   * receiving broadcasts. By default, this will only wait for the amount of
+   * specified via the `deferNoticationTimeout` option.
    *
-   * @param {Response} firstResponse First responses to compare.
-   * @param {Response} secondResponse Second responses to compare.
-   * @param {string} url The URL of the updated request.
-   * @param {string} cacheName Name of the cache the responses belong to.
-   * This is included in the message posted on the broadcast channel.
+   * @private
    */
-  notifyIfUpdated(firstResponse, secondResponse, url, cacheName) {
-    if (!responsesAreSame(
-      firstResponse, secondResponse, this._headersToCheck)) {
-      broadcastUpdate(this._getChannel(), cacheName, url, this._source);
-    }
+  async _windowReadyOrTimeout() {
+    await new Promise((resolve) => {
+      let timeout;
+
+      // Set a callback so that if a message comes in within the next
+      // few seconds the promise will resolve.
+      this._onReadyMessageCallback = () => {
+        this._onReadyMessageCallback = null;
+        clearTimeout(timeout);
+        resolve();
+      };
+
+      // But don't wait too long for the message since it may never come.
+      timeout = setTimeout(() => {
```

Comment:
I can do that. I think I did it in the other order because I was referencing timeout in a callback before assigning it (and I was referencing it first so the comments made sense and were in sequential order).